lib/client.js

```diff
@@ -131,9 +151,17 @@ function createError (error) {
   return err
 }
 
-function removePromise (id, timeoutId, args) {
+function removePromise (id, timeoutId, client, actions, promiseStartTime, args) {
   clearTimeout(timeoutId)
   delete pendingPromises[id]
+
+  if (promiseStartTime) { // only track time for successfully resolved promises
```

Comment:
⛏ may be we move this logic to another function so that `removePromise` remains pure?

Reply:
Not sure I understand. Wouldn't extracting this function (and presumably the additional `client`, `actions`, `promiseStartTime` params) make this _impure_?

Reply:
Yea what I meant is, what if we just move the whole thing out and not even bother passing those arguments.

I might need to spend more time understanding how all the things are hooked up together here but I was thinking something like this would do it;

```
const trackLongRequestsCallback = trackLongRequests.bind(client, actions, promiseStartTime)
promise
  .then(removePromise.bind(null, id, timeoutId), removePromise.bind(null, id, timeoutId))
  .then(trackLongRequestsCallback)
  .catch(trackLongRequestsCallback)
```

`catch` is if we want to catch rejected requests too (which I assume we want to?).

Reply:
We don't want to catch rejected requests, but otherwise I agree this seems like a better approach 👍

Reply:
...actually this won't work since it changes the value returned from `wrappedPostMessage` and  passed to `processResponse` in handlers like `get`, e.g. https://github.com/zendesk/zendesk_app_framework_sdk/pull/88/files#diff-50cfa59973c04321b5da0c6da0fdf4feR596

Reply:
ic, I guess we can just leave this for now then.